drivers/adc-dac/ad5592r/ad5592r-base.c

```diff
@@ -320,3 +320,184 @@ int32_t ad5592r_reset_channel_modes(struct ad5592r_dev *dev)
 
 	return ad5592r_set_channel_modes(dev);
 }
+
+/**
+ * Register update
+ *
+ * @param dev - The device structure.
+ * @param reg_addr - The Register address
+ * @param data - The data to be written
+ * @param mask - The mask
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_base_reg_update(struct ad5592r_dev *dev, uint16_t reg_addr,
+				uint16_t data, uint16_t mask)
+{
+
+	uint16_t temp_reg_val;
+	int32_t ret;
+
+	if (!dev)
+		return -1;
+
+	ret = ad5592r_base_reg_read(dev, reg_addr, &temp_reg_val);
+	if (ret < 0)
+		return ret;
+
+	temp_reg_val &= ~mask;
+	temp_reg_val |= data;
+
+	ret = ad5592r_base_reg_write(dev, reg_addr, temp_reg_val);
+	if (ret < 0)
+		return ret;
+
+	return 0;
+}
+
+/**
+ * Set ADC Range of the device
+ *
+ * @param dev - The device structure.
+ * @param adc_range - ADC Range
+ * @return 0 in case of success, negative error code otherwise
+ */
+int32_t ad5592r_set_adc_range(struct ad5592r_dev *dev,
+			      enum ad559xr_range adc_range)
+{
+	int32_t ret;
+	uint16_t status;
+
+	if (!dev)
+		return -1;
+
+	if (adc_range)
+		status |= AD5592R_REG_CTRL_ADC_RANGE;
+	else
+		status &= ~AD5592R_REG_CTRL_ADC_RANGE;
```

Comment:
you can rewrite this whole part more compact as:
```
uint16_t status = adc_range ? AD5592R_REG_CTRL_ADC_RANGE : 0;
```

or even ditch the status local variable and put it all inline in the function call, although that's maybe a bit too much